### PR TITLE
Add support for @Resource() missing type parameter quick fix for Jakarta

### PR DESCRIPTION
**Pull request by turkeylurkey**
In one or two cases I reformatted a lambda so that I could insert LOG stmts.

### CODE REVIEW

File: src/main/java/io/openliberty/tools/intellij/lsp4mp/lsp4ij/operations/codeactions/LSPCodeActionAnnotator.java [hunk 1]
```diff
@@ -132,9 +132,14 @@ protected Info doCollectInformation(PsiFile file, Editor editor) {
                     params.setContext(context);
                     params.setTextDocument(new TextDocumentIdentifier(LSPIJUtils.toUri(collectedInfo.file).toString()));
                     params.setRange(diagnostic.getRange());
-                    CompletableFuture<List<Either<Command, CodeAction>>> codeAction = entry.getKey().getInitializedServer().thenComposeAsync(server -> server.getTextDocumentService().codeAction(params));
+                    LanguageServerWrapper wrap = entry.getKey();
```
Comment (mrglavas) [inline src/main/java/io/openliberty/tools/intellij/lsp4mp/lsp4ij/operations/codeactions/LSPCodeActionAnnotator.java:135]: Did you mean to use this variable somewhere? Maybe on the next line? I don't see any references.

File: src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/codeAction/JakartaCodeActionHandler.java [hunk 2]
```diff
@@ -57,8 +57,8 @@ public List<CodeAction> codeAction(JakartaJavaCodeActionParams params, IPsiUtils
             int start = DiagnosticsHelper.getStartOffset(unit, params.getRange(), utils);
             int end = DiagnosticsHelper.getEndOffset(unit, params.getRange(), utils);
             var mpParams = new MicroProfileJavaCodeActionParams(params.getTextDocument(), params.getRange(), params.getContext());
-            JavaCodeActionContext context = new JavaCodeActionContext(unit, start, end - start, utils,
-                    mpParams);
+            JavaCodeActionContext context = new JavaCodeActionContext(unit.getViewProvider().clone().getPsi(unit.getLanguage()),
```
Comment (mrglavas) [inline src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/codeAction/JakartaCodeActionHandler.java:60]: Might be wise to add a comment here on the importance of cloning the PsiFile.

File: src/main/java/io/openliberty/tools/intellij/lsp4mp/lsp4ij/operations/codeactions/LSPCodeActionAnnotator.java (hunk 1 above)
Comment (turkeylurkey) [reply src/main/java/io/openliberty/tools/intellij/lsp4mp/lsp4ij/operations/codeactions/LSPCodeActionAnnotator.java:135]: You're right, that was part of a log statement. I'll take it out.

File: src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/codeAction/JakartaCodeActionHandler.java (hunk 2 above)
Comment (evie-lau) [reply src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/codeAction/JakartaCodeActionHandler.java:60]: it seems JavaCodeActionContext already has a `.copy()` method that we can use, which uses this same method of cloning the `PsiFile.getViewProvider()`

Comment (turkeylurkey) [reply src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/codeAction/JakartaCodeActionHandler.java:62]: Good idea.

File: src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/codeAction/proposal/NewAnnotationProposal.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
+/*******************************************************************************
+ * Copyright (c) 2000, 2021 IBM Corporation and others.
```
Comment (evie-lau) [inline src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/codeAction/proposal/NewAnnotationProposal.java:2]: 2023 copyright year, same for the other files

**Review by evie-lau — APPROVED** [review]

File: src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/codeAction/proposal/NewAnnotationProposal.java (hunk 3 above)
Comment (turkeylurkey) [reply src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/codeAction/proposal/NewAnnotationProposal.java:2]: 👍

**Pull request merged**